src/util/files.cpp

```diff
@@ -29,4 +30,13 @@ get_root_path() {
     return root_path;
 }
 
+std::optional<std::ifstream>
+open_file(std::filesystem::path path) {
+    if (!std::filesystem::exists(path)) {
+        LOG_ERROR(logging::file_io_logger, "File, {}, not found", path.string());
+        return {};
+    }
+    return std::ifstream(path);
```

Comment:
```suggestion
    if (!std::filesystem::exists(path)) [[unlikely]] {
        LOG_WARNING(logging::file_io_logger, "File `{}` not found", path.string());
        return {};
    }
    
    return std::ifstream(path);
```

put this in the header so it can be inlined

Reply:
I did initially, but the logger can't be defined in file. The logging header includs the file header.

Reply:
wahtever